### ISSUE TITLE
Follow  some more fine grained approach for applying the tabular fonts class  from Patternfly

### ISSUE THREAD

**Issue by KKoukiou** [cockpit-project/cockpit#18125]
We now insert this class on the  top of all page, thus making anything numeric (there are exceptions ?!) use tabular fonts. We should rather focus  on adding this only to elements that need this. For example sliders values.

**Comment by martinpitt** [comment]
Big +1. IMHO this looks super-ugly right now, as the monospaced text is being used overzealously everywhere. We have wild font inconsistencies like this now:

![image](https://user-images.githubusercontent.com/200109/210770096-cf09e1a3-5efd-4188-a175-dab6f4e8b29e.png)

![image](https://user-images.githubusercontent.com/200109/210770135-a31a959c-35d7-4b47-8e62-d7c0d30cd114.png)

Notice how we format zeros differently in the Configuration card, and  how ugly the dates are.

Monospaced numbers only make sense in some tabular environments where they *actually* need to be aligned. The only place where this is the case on the overview page is here:

![image](https://user-images.githubusercontent.com/200109/210770342-13c1c5bb-6394-4d3e-aa5e-4402091c81d8.png)

But ironically it doesn't even help, as the progress bars still have different lengths.

I went through all the pages, and the only other potential usages of tabular numbers that I found are on hwinfo:

![image](https://user-images.githubusercontent.com/200109/210770534-e795d661-81e2-4eaa-9282-599b7541a00f.png)

(Although it's not really necessary there)

and to ensure properly aligned timestamps on the Networking page's Log view:

![image](https://user-images.githubusercontent.com/200109/210770620-c282851d-6727-461f-bf65-310c30852e02.png)

But in all of these cases we'd actually be better off with using a proper monospace font.

In other words: These need to be opt-in and carefully/explicitly used, instead of "everywhere". They are called *tabular* for a reason? 

Also, I find these zeros super ugly, at first I thought my system was broken and somehow missed proper fonts -- but that's a matter of taste. But more objectively, paragraph text or most text which doesn't rely on alignment should use proportional fonts.

**Comment by garrett** [comment]
The 0 with a slash is a bug in the font; it should have variants with and without the slash, and it should be controlled by a font feature (which is exposed in CSS).

Tabular numbers are needed for columns in tables that have numbers and anywhere where numbers are compared, including numbers compared with themselves over time (such as progress bars and sliders) and numbers compared with other numbers (which is why I mentioned columns in tables). It's also useful for things like IP addresses.

Tabular numbers are _not_ needed for a mix of alphanumeric instead of just numeric where it's not being compared or changing over time.

So we really have 3 issues:

1. Don't apply everywhere.
2. Apply in circumstances listed above
   - number comparison, numbers that change, numbers with units, IP addresses, etc.
3. Fix the font so it doesn't use 0 with a slash.
   - Which may require me doing a hotfix to the font with a monkeypatch in the meantime, which we've done before, as the font is quite slow to update and propagate through PatternFly to us. This would be done via font subsetting and adding a font to a typeface for a specific unicode range (in this case: one character).
      - MDN has info on changing a single character: https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/unicode-range#examples
      - We'd use a data URI in the `@font-face` specification for the tabular data class, like `url(data:font/woff;charset=utf-8;base64,BASE64_ENCODED_DATA_HERE) format(‘woff’)`
   - We actually want `font-variant-numeric: tabular-nums;` not `font-variant-numeric: tabular-nums slashed-zero;` (which is what we're seeing) https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-numeric#values
   - I seem to remember that Red Hat Mono doesn't have the slash, but the tabular version in Red Hat Text does? If I'm not wrong, then we don't even have to subset; we can just re-use the font for just that one character. Then it's just a sprinkling of CSS to fix the problem.

**Comment by garrett** [comment]
> But ironically it doesn't even help fully, as the progress bars still have different lengths. It does help to not make the memory bar "jumpy" if the used amount changes a little, though.

No, that's another bug. It's properly solved with `subgrid` or `display: contents`, but we've been waiting on all browsers to catch up. (I really mean we've been waiting for years for _Chrome_, specifically. Firefox had this for years and even Safari has had support for a little while already.)

**Comment by garrett** [comment]
We also need to report this upstream to both PatternFly and RedHatFont.

**Comment by garrett** [comment]
I've filed upstream @:

- https://github.com/RedHatOfficial/RedHatFont/issues/53

- https://github.com/patternfly/patternfly/issues/5308

**Comment by garrett** [comment]
RedHatMono also seems to have the slash. I guess this also needs a patch. And I guess this means I'll have to do type "surgery" in FontForge again with a subset for a hotfix. :disappointed: (And for the various variants as well, such as bold and possibly italic if there's a mono form.)

Hopefully the metrics are the same between RedHatMono and RedHatText and we can just use the same subset for both.

**Comment by garrett** [comment]
I've made a CSS patch locally, to remove the slash!

Before:

![image](https://user-images.githubusercontent.com/10246/211325114-63616473-d877-448c-90df-08150f0c9af7.png)



After:

![image](https://user-images.githubusercontent.com/10246/211324327-e0d792a5-8fe5-4e3d-8d24-c92733548650.png)

As this is working, the plan would be to proceed by modifying all the font variants (various weights & italic versions of each).

**Comment by garrett** [comment]
Here's an image diff; it shows that the metrics are untouched; the only difference really is the slash. (There's a little bit of hinting shift in the glyph, but it's sub-pixel. The usage bars are slightly different with values.)

![image](https://user-images.githubusercontent.com/10246/211327138-af25ecf9-bc30-4074-be8a-e9dfcad3dc70.png)
